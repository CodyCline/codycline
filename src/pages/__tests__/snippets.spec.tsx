import React from "react"
import renderer from "react-test-renderer";
import { getPageQueryData } from 'gatsby-plugin-testing';
import Snippets from "../snippets";

describe("Snippets", () => {
    it("renders GraphQL data correctly", async() => {
        const data = await getPageQueryData(`snippets`);
        const tree = renderer.create(<Snippets data={data} />).toJSON();
        expect(tree).toMatchSnapshot()
    })
})