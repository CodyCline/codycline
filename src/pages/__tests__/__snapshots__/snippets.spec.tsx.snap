// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snippets renders GraphQL data correctly 1`] = `
Array [
  <nav
    className="navbar"
  >
    <a
      className="nav__item home__icon"
      href="/"
    >
      <img
        alt="logo.png"
        src="test-file-stub"
        style={
          Object {
            "height": "50px",
            "width": "70px",
          }
        }
      />
    </a>
    <a
      className="nav__item"
      href="/blog"
    >
       Blog 
    </a>
    <a
      className="nav__item"
      href="/snippets"
    >
      Snippets
    </a>
    <a
      className="nav__item"
      href="/projects"
    >
      Projects
    </a>
    <a
      className="nav__item controls"
      href="#"
    />
  </nav>,
  <div
    style={
      Object {
        "height": "10vh",
        "width": "undefinedvh",
      }
    }
  />,
  <div
    className="content__container"
  >
    <div
      className="content__header"
    >
      <h2
        className="no__margin"
      >
        Snippets
      </h2>
      <p>
        Bite-sized content for quick reading and implementation.
      </p>
      <hr
        className="divider"
      />
    </div>
    <ul
      className="strip"
    >
      <li
        className="strip__title"
      >
        <span
          className="strip__icon"
        >
          <svg
            height={32}
            viewBox="0 0 1024 1024"
            width={32}
          >
            <title>
              rust
            </title>
            <path
              d="M483.968 169.6c0-14.704 11.92-26.624 26.624-26.624s26.624 11.92 26.624 26.624c0 14.704-11.92 26.624-26.624 26.624v0c-14.704-0-26.624-11.92-26.624-26.624v0zM166.4 410.688c0-14.704 11.92-26.624 26.624-26.624s26.624 11.92 26.624 26.624c0 14.704-11.92 26.624-26.624 26.624v0c-14.704 0-26.624-11.92-26.624-26.624v0zM801.792 411.936c0-14.704 11.92-26.624 26.624-26.624s26.624 11.92 26.624 26.624c0 14.704-11.92 26.624-26.624 26.624v0c-14.704 0-26.624-11.92-26.624-26.624v0zM243.36 448.416c8.563-3.889 14.41-12.367 14.41-22.209 0-3.543-0.758-6.909-2.12-9.945l0.062 0.154-11.808-26.72h46.464v209.44h-93.76c-7.957-26.904-12.536-57.815-12.536-89.795 0-12.455 0.695-24.749 2.047-36.843l-0.135 1.486zM437.728 453.568v-61.728h110.656c5.728 0 40.352 6.592 40.352 32.512 0 21.504-26.56 29.216-48.416 29.216zM286.656 785.952c0-14.704 11.92-26.624 26.624-26.624s26.624 11.92 26.624 26.624c0 14.704-11.92 26.624-26.624 26.624v0c-14.704 0-26.624-11.92-26.624-26.624v0zM681.248 787.2c0-14.704 11.92-26.624 26.624-26.624s26.624 11.92 26.624 26.624c0 14.704-11.92 26.624-26.624 26.624v0c-14.704 0-26.624-11.92-26.624-26.624v0zM689.472 726.816c-1.517-0.337-3.259-0.53-5.047-0.53-11.618 0-21.331 8.158-23.723 19.057l-0.030 0.161-13.376 62.368c-39.739 18.451-86.258 29.214-135.29 29.214-50.139 0-97.649-11.254-140.142-31.375l1.993 0.849-13.344-62.368c-2.419-11.046-12.119-19.194-23.722-19.194-1.799 0-3.552 0.196-5.239 0.567l0.161-0.030-55.072 11.84c-9.931-10.217-19.266-21.166-27.861-32.699l-0.619-0.869h267.968c3.040 0 5.056-0.544 5.056-3.2v-94.912c0-2.752-2.016-3.2-5.056-3.2h-78.4v-60.192h84.768c26.346 0.272 48.091 19.625 52.089 44.882l0.039 0.302c3.36 13.216 10.752 56.224 15.808 69.984 5.024 15.456 25.6 46.304 47.424 46.304h138.336c-9.735 12.981-19.688 24.466-30.381 35.213l0.013-0.013zM838.272 476.544c1.028 9.746 1.615 21.056 1.615 32.503 0 8.596-0.331 17.114-0.98 25.543l0.069-1.118h-33.632c-3.36 0-4.736 2.208-4.736 5.504v15.456c0 36.352-20.512 44.288-38.4 46.304-17.12 1.92-36.096-7.168-38.4-17.632-2.653-37.842-23.132-70.395-53.012-89.579l-0.46-0.277c32.96-20.928 67.2-51.808 67.2-93.152-1.504-36.699-21.557-68.403-50.982-86.135l-0.474-0.265c-19.951-13.093-44.044-21.505-69.977-23.147l-0.423-0.021h-347.936c46.95-52.364 110.031-89.409 181.387-103.312l2.069-0.336 41.024 43.008c4.427 4.643 10.66 7.53 17.567 7.53 6.51 0 12.421-2.565 16.778-6.738l-0.009 0.008 45.888-43.904c96.756 18.604 176.937 77.623 223.637 158.42l0.843 1.58-31.424 70.944c-1.294 2.878-2.048 6.238-2.048 9.774 0 9.845 5.842 18.325 14.248 22.163l0.152 0.062zM916.608 477.696l-1.056-10.976 32.352-30.176c2.312-2.415 3.735-5.697 3.735-9.311 0-5.468-3.258-10.175-7.938-12.287l-0.085-0.034-41.216-15.456-3.2-10.688 25.792-35.84c1.312-2.042 2.092-4.535 2.092-7.21 0-6.39-4.448-11.74-10.417-13.124l-0.091-0.018-43.616-7.104-5.248-9.792 18.336-40.224c0.594-1.461 0.939-3.155 0.939-4.93 0-7.319-5.864-13.268-13.151-13.406l-0.013-0-44.256 1.536-7.168-8.448 10.176-43.104c0.127-0.695 0.2-1.495 0.2-2.312 0-7.44-6.032-13.472-13.472-13.472-0.817 0-1.617 0.073-2.394 0.212l0.082-0.012-43.136 10.080-8.512-7.008 1.568-44.192c-0.178-7.302-6.14-13.154-13.468-13.154-1.739 0-3.401 0.33-4.927 0.929l0.091-0.032-40.224 18.336-9.792-5.248-7.104-43.616c-1.402-6.060-6.752-10.508-13.142-10.508-2.675 0-5.168 0.78-7.264 2.124l0.054-0.032-35.872 25.792-10.656-3.2-15.456-41.376c-2.145-4.766-6.853-8.024-12.322-8.024-3.599 0-6.868 1.411-9.284 3.71l0.006-0.005-30.176 32.384-10.976-1.056-23.296-37.664c-2.47-3.47-6.478-5.705-11.008-5.705s-8.538 2.236-10.98 5.664l-0.028 0.041-23.296 37.664-10.976 1.056-30.176-32.384c-2.411-2.294-5.68-3.704-9.278-3.704-5.469 0-10.176 3.258-12.287 7.94l-0.034 0.085-15.584 41.184-10.656 3.2-35.84-25.6c-2.042-1.312-4.535-2.092-7.21-2.092-6.39 0-11.74 4.448-13.124 10.417l-0.018 0.091-7.104 43.616-9.792 5.248-40.256-18.336c-1.443-0.577-3.116-0.912-4.867-0.912-7.327 0-13.285 5.863-13.437 13.154l-0 0.014 1.536 44.256-8.512 6.944-43.104-10.112c-0.695-0.127-1.495-0.2-2.312-0.2-7.44 0-13.472 6.032-13.472 13.472 0 0.817 0.073 1.617 0.212 2.394l-0.012-0.082 10.112 43.136-6.976 8.48-44.224-1.536c-7.305 0.197-13.152 6.165-13.152 13.499 0 1.739 0.329 3.402 0.928 4.928l-0.032-0.092 18.336 40.224-5.248 9.792-43.616 7.104c-6.060 1.402-10.508 6.752-10.508 13.142 0 2.675 0.78 5.168 2.124 7.264l-0.032-0.054 25.792 35.84-3.2 10.688-41.376 15.456c-4.766 2.146-8.023 6.853-8.023 12.321 0 3.614 1.423 6.896 3.74 9.316l-0.005-0.005 32.352 30.176-1.056 10.976-37.664 23.296c-3.47 2.47-5.705 6.478-5.705 11.008s2.236 8.538 5.664 10.98l0.041 0.028 37.664 23.296 1.056 10.976-32.352 30.176c-2.299 2.412-3.714 5.685-3.714 9.288 0 5.46 3.248 10.161 7.917 12.277l0.085 0.034 41.376 15.456 3.2 10.688-25.792 35.872c-1.31 2.041-2.088 4.531-2.088 7.204 0 6.4 4.462 11.757 10.445 13.131l0.091 0.018 43.616 7.104 5.248 9.824-18.336 40.224c-0.576 1.444-0.91 3.116-0.91 4.867 0 7.333 5.858 13.297 13.15 13.468l0.016 0 44.256-1.536 7.008 8.512-10.144 43.136c-0.128 0.697-0.201 1.499-0.201 2.318 0 7.423 6.017 13.44 13.44 13.44 0.829 0 1.64-0.075 2.428-0.219l-0.082 0.012 43.040-10.208 8.512 6.976-1.568 44.224c0.17 7.291 6.121 13.136 13.437 13.136 1.751 0 3.424-0.335 4.958-0.944l-0.091 0.032 40.224-18.336 9.792 5.248 7.104 43.584c1.391 6.074 6.748 10.536 13.148 10.536 2.672 0 5.163-0.778 7.258-2.12l-0.054 0.032 35.84-25.824 10.688 3.2 15.456 41.344c2.15 4.754 6.852 8.002 12.312 8.002 3.603 0 6.876-1.415 9.294-3.719l-0.005 0.005 30.176-32.352 10.976 1.088 23.296 37.664c2.478 3.457 6.483 5.682 11.008 5.682s8.53-2.225 10.98-5.642l0.028-0.041 23.296-37.664 10.976-1.088 30.176 32.352c2.412 2.299 5.685 3.714 9.288 3.714 5.46 0 10.161-3.248 12.277-7.917l0.034-0.085 15.456-41.344 10.688-3.2 35.84 25.824c2.041 1.31 4.531 2.088 7.204 2.088 6.4 0 11.757-4.462 13.131-10.445l0.018-0.091 7.104-43.584 9.792-5.248 40.224 18.336c1.443 0.577 3.116 0.912 4.867 0.912 7.316 0 13.267-5.845 13.436-13.121l0-0.016-1.536-44.288 8.48-6.976 43.104 10.144c0.705 0.131 1.517 0.206 2.345 0.206 7.423 0 13.44-6.017 13.44-13.44 0-0.819-0.073-1.621-0.214-2.4l0.012 0.082-9.984-43.072 6.976-8.512 44.256 1.536c7.294-0.189 13.134-6.146 13.134-13.468 0-1.751-0.334-3.425-0.942-4.959l0.032 0.091-18.336-40.224 5.248-9.824 43.616-7.104c6.067-1.381 10.527-6.73 10.527-13.121 0-2.685-0.787-5.185-2.143-7.284l0.032 0.053-25.792-35.84 3.2-10.688 41.376-15.456c4.765-2.136 8.025-6.838 8.025-12.301 0-3.611-1.424-6.889-3.741-9.304l0.004 0.005-32.352-30.176 1.056-10.976 37.664-23.296c3.47-2.47 5.705-6.478 5.705-11.008s-2.236-8.538-5.664-10.98l-0.041-0.028z"
              fill={null}
              opacity={1}
            />
          </svg>
        </span>
        <a
          href="/snippets/rust-blockchain/"
        >
          Markdown Test
        </a>
      </li>
      <li
        className="strip__description"
      >
        <p>
          Build a blockchain in rust
        </p>
      </li>
      <li
        className="strip__date"
      >
        <p>
          1970-01-01
        </p>
      </li>
    </ul>
    <ul
      className="strip"
    >
      <li
        className="strip__title"
      >
        <span
          className="strip__icon"
        >
          <svg
            height={32}
            viewBox="0 0 1024 1024"
            width={32}
          >
            <title>
              graphql
            </title>
            <path
              d="M153.013 727.89l366.427-634.645 32.895 18.993-366.427 634.645-32.895-18.993z"
              fill="rgb(225, 0, 152)"
              opacity={1}
            />
            <path
              d="M145.44 677.184h732.864v37.984h-732.864v-37.984z"
              fill="rgb(225, 0, 152)"
              opacity={1}
            />
            <path
              d="M159.948 698.519l18.993-32.895 366.554 211.638-18.993 32.895-366.554-211.638z"
              fill="rgb(225, 0, 152)"
              opacity={1}
            />
            <path
              d="M478.416 146.919l18.993-32.895 366.554 211.638-18.993 32.895-366.554-211.638z"
              fill="rgb(225, 0, 152)"
              opacity={1}
            />
            <path
              d="M160.047 325.555l366.554-211.638 18.993 32.895-366.554 211.638-18.993-32.895z"
              fill="rgb(225, 0, 152)"
              opacity={1}
            />
            <path
              d="M471.768 112.24l32.895-18.993 366.427 634.645-32.895 18.993-366.427-634.645z"
              fill="rgb(225, 0, 152)"
              opacity={1}
            />
            <path
              d="M174.528 300.352h37.984v423.296h-37.984v-423.296z"
              fill="rgb(225, 0, 152)"
              opacity={1}
            />
            <path
              d="M811.52 300.352h37.984v423.296h-37.984v-423.296z"
              fill="rgb(225, 0, 152)"
              opacity={1}
            />
            <path
              d="M503.468 865.35l318.805-184.070 16.593 28.738-318.805 184.070-16.593-28.738z"
              fill="rgb(225, 0, 152)"
              opacity={1}
            />
            <path
              d="M899.84 736c-14.077 24.135-39.843 40.097-69.338 40.097-44.183 0-80-35.817-80-80s35.817-80 80-80c14.751 0 28.57 3.992 40.434 10.956l-0.376-0.204c24.049 14.097 39.942 39.817 39.942 69.249 0 14.688-3.959 28.452-10.867 40.283l0.205-0.379z"
              fill="rgb(225, 0, 152)"
              opacity={1}
            />
            <path
              d="M262.4 368c-14.077 24.135-39.843 40.097-69.338 40.097-44.183 0-80-35.817-80-80s35.817-80 80-80c14.751 0 28.57 3.992 40.434 10.956l-0.376-0.204c24.049 14.097 39.942 39.817 39.942 69.249 0 14.688-3.959 28.452-10.867 40.283l0.205-0.379z"
              fill="rgb(225, 0, 152)"
              opacity={1}
            />
            <path
              d="M124.16 736c-6.759-11.488-10.751-25.307-10.751-40.058 0-44.183 35.817-80 80-80s80 35.817 80 80c0 29.494-15.961 55.261-39.717 69.133l-0.379 0.205c-11.451 6.704-25.215 10.662-39.903 10.662-29.432 0-55.151-15.893-69.044-39.566l-0.204-0.376z"
              fill="rgb(225, 0, 152)"
              opacity={1}
            />
            <path
              d="M761.6 368c-6.759-11.488-10.751-25.307-10.751-40.058 0-44.183 35.817-80 80-80s80 35.817 80 80c0 29.494-15.961 55.261-39.717 69.133l-0.379 0.205c-11.451 6.704-25.215 10.662-39.903 10.662-29.432 0-55.151-15.893-69.044-39.566l-0.204-0.376z"
              fill="rgb(225, 0, 152)"
              opacity={1}
            />
            <path
              d="M512 960c-44.183 0-80-35.817-80-80s35.817-80 80-80c44.183 0 80 35.817 80 80v0c0 0.067 0 0.145 0 0.224 0 44.059-35.717 79.776-79.776 79.776-0.079 0-0.158-0-0.237-0l0.012 0z"
              fill="rgb(225, 0, 152)"
              opacity={1}
            />
            <path
              d="M512 223.712c-44.183 0-80-35.817-80-80s35.817-80 80-80c44.183 0 80 35.817 80 80v0c0 0.067 0 0.145 0 0.224 0 44.059-35.717 79.776-79.776 79.776-0.079 0-0.158-0-0.237-0l0.012 0z"
              fill="rgb(225, 0, 152)"
              opacity={1}
            />
          </svg>
        </span>
        <a
          href="/snippets/test-snippet/"
        >
          Markdown Test
        </a>
      </li>
      <li
        className="strip__description"
      >
        <p>
          Some really cool content  that needs to be.
        </p>
      </li>
      <li
        className="strip__date"
      >
        <p>
          1970-01-01
        </p>
      </li>
    </ul>
  </div>,
  <div
    style={
      Object {
        "height": "60vh",
        "width": "undefinedvh",
      }
    }
  />,
  <footer
    className="footer"
  >
    <li
      className="footer__item"
    >
      2021
    </li>
    <li
      className="footer__item"
    />
  </footer>,
]
`;
